### PR TITLE
Add chunk_while

### PR DESCRIPTION
**Pull request by jsyeo**

### CODE REVIEW

File: kernel/common/enumerable.rb [hunk 1]
```diff
@@ -212,6 +212,12 @@ def slice_when(&block)
     end
   end
 
+  def chunk_while(&block)
+    raise ArgumentError, "wrong number of arguments (0 for 1)" unless block_given?
```
Comment (jemc) [inline kernel/common/enumerable.rb]: This doesn't strike me as a particularly accurate or helpful error message.

Even if it's the same as the one given by MRI, I'd prefer to see it made clear that a *block is required*.  The current error message makes it look like a regular argument is required.

Comment (brixen) [inline kernel/common/enumerable.rb]: Hm, interesting that this doesn't return an Enumerator when no block is given. I see this in the 2.3.0 `ri chunk_while` docs:

```
= .chunk_while

(from ruby core)
=== Implementation from Enumerable
------------------------------------------------------------------------------
  enum.chunk_while {|elt_before, elt_after| bool } -> an_enumerator

------------------------------------------------------------------------------
```

Assuming this should raise an ArgumentError, I agree with @jemc here, we should use a more helpful error message.

Comment (brixen) [inline kernel/common/enumerable.rb]: Ah, I see now why the block is required. So yeah, we should fix the error message.

Comment (jsyeo) [inline kernel/common/enumerable.rb]: What should the error be? How about "Block required"?

Comment (jsyeo) [inline kernel/common/enumerable.rb]: Or how about "no block given" for consistency with `chunk` in https://github.com/rubinius/rubinius/blob/master/kernel/common/enumerable.rb#L8

**Comment by jsyeo** [comment]
Changed error message to "no block given".

File: kernel/common/enumerable.rb (hunk 1 above)
Comment (jemc) [inline kernel/common/enumerable.rb]: :+1:

File: kernel/common/enumerable.rb [hunk 2]
```diff
@@ -212,6 +212,12 @@ def slice_when(&block)
     end
   end
 
+  def chunk_while(&block)
+    raise ArgumentError, "no block given" unless block_given?
+
+    slice_when { |before, after| !(block.call(before, after)) }
```
Comment (YorickPeterse) [inline kernel/common/enumerable.rb]: The explicit block isn't needed here, this can just be:

    !(yield before, after)

**Comment by brixen** [comment]
Thanks!

**Pull request merged**